projects/solar/src/mppt.c

```diff
@@ -0,0 +1,42 @@
+#include "gpio.h"
+#include "mux.h"
+#include "spv1020_mppt.h"
+
+// will get the address and size of the demultiplexer
+static MuxAddress s_mux_address = {
+  .bit_width = 3,
+  .sel_pins[0] = { .port = GPIO_PORT_B, .pin = 3 },
+  .sel_pins[1] = { .port = GPIO_PORT_B, .pin = 4 },
+  .sel_pins[2] = { .port = GPIO_PORT_B, .pin = 5 },
+  .mux_enable_pin = { .port = GPIO_PORT_A, .pin = 1 },  // need to check hardward for correct pin
+};
+
+StatusCode mppt_shut(SpiPort port, uint8_t pin) {
+  status_ok_or_return(mux_set(&s_mux_address, pin));
+  return spv1020_shut(port);
```

Comment:
Like I explained on Slack, since hardware hasn't given us an enable pin, to set the CS line for SPI low and then high again for the specific SPV1020 we're talking to, we have to set the mux to the `pin` before calling the `spv1020` function, then set the mux to an output we're not using (like output 7). That way, the SPV1020 sees the CS line go low, then high again at the end.

Thus, you should add a `#define DISCONNECTED_MUX_OUTPUT 7`, then call `mux_set(&s_mux_address, DISCONNECTED_MUX_OUTPUT)` at the end of each of the `mppt_*` functions.

Also, to deal with status codes: to ensure that the final `mux_set` call you're adding is always called even if the `spv1020` call fails, you should save the output of the `spv1020_*` call in a `status` variable, wrap the final `mux_set` call in `status_ok_or_return`, then return the `status` variable you saved. That way the final `mux_set` is always executed and we never end up with the CS line low for a long period of time.